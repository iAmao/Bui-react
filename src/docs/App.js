import React from 'react';
import PropTypes from 'prop-types';

import Routes from './Routes';
import Router from './Router';
import NotFound from './NotFound';
import Component from './Component';
import GetStarted from './GetStarted';
import Navigation from './Navigation';

import componentData from '../../config/componentData';

import BuiTheme from '../components/BuiTheme'


const PATH = window.location.pathname.substr(1);


class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = { route: props.location || PATH };
    }

    render() {
        const componentMap = componentData.reduce((a, component) => {
            a[component.name.toLowerCase()] = component;
            return a;
        }, {});

        return (
            <Router render={(location) => {
                return (
                    <div className="row">
                        <div className="col-md-2" id="navigation-container">
                            <Navigation
                                path={location.path}
                                components={componentData.map((component) => component.name)}
                            />
                        </div>
                        <div className="col-md-8" id="details-container">
                            <BuiTheme>
                                <Routes
                                    routes={(location) => {
                                        return [
                                            ['/', GetStarted],
                                            ['/theming', () => <div><h2>Theming</h2></div>],
                                            ['/:id', (props) => {
                                                const component = componentMap[props.location.params.id];
                                                return component ?
                                                    <Component component={componentMap[props.location.params.id]} />
                                                    :
                                                    <NotFound />
                                            }],
                                            ['*', NotFound]
                                        ]
                                    }}
                                />
                            </BuiTheme>
                        </div>
                    </div>
                );
            }} />
        )
    }
}

App.propTypes = {
    location: PropTypes.string
};

export default App;
