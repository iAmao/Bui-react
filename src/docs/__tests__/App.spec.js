import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { mount } from 'enzyme';

import App from '../App';


describe('DOCS: App', () => {

    describe('Snapshot Test', () => {
        it('renders the "Getting Started" page for url "/"', () => {
            const tree = renderer.create(<App />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders the "Theming" page for url "/theming"', () => {
            window.history.pushState(null, '', '/theming')
            const tree = renderer.create(<App />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders the "Component" page for url "/:id"', () => {
            window.history.pushState(null, '', '/button')
            const tree = renderer.create(<App />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders 404 page if a valid route is not found', () => {
            window.history.pushState(null, '', '/onlyone')
            const tree = renderer.create(<App location="lol" />).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('Integration Test', () => {
        it('renders documentation and examples', () => {
            const div = document.createElement('div');
            ReactDOM.render(<App />, div);
            ReactDOM.unmountComponentAtNode(div);
        });

        it('renders not found page on non-existent components', () => {
            window.history.pushState(null, '', '/random/page')
            const wrapper = mount(<App />);

            expect(wrapper.find({ id: 'bui-react-docs-component-not-found' })).toHaveLength(1);
            expect(wrapper.find({ id: 'bui-react-docs-component-not-found' }).children().text()).toEqual('404!');
        });
    });
});
