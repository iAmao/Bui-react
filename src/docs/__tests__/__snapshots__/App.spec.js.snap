// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOCS: App Snapshot Test renders 404 page if a valid route is not found 1`] = `
<div
  className="row"
>
  <div
    className="col-md-2"
    id="navigation-container"
  >
    <div>
      <div
        id="navigation-logo"
      >
        <h4>
          bui React
        </h4>
      </div>
      <div
        id="navigation-items"
      >
        <ul
          className="navigation"
        >
          <li
            className="bui-react-docs-component-name "
          >
            <a
              onClick={[Function]}
            >
              Getting started
            </a>
          </li>
          <li
            className="bui-react-docs-component-name "
          >
            <a
              onClick={[Function]}
            >
              Theming
            </a>
          </li>
          <li
            className="bui-react-docs-component-name "
          >
            <a
              onClick={[Function]}
            >
              Button
            </a>
          </li>
          <li
            className="bui-react-docs-component-name "
          >
            <a
              onClick={[Function]}
            >
              Icon
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="col-md-8"
    id="details-container"
  >
    <div
      id="bui-react-docs-component-not-found"
    >
      <h3>
        404!
      </h3>
    </div>
  </div>
</div>
`;

exports[`DOCS: App Snapshot Test renders the "Component" page for url "/:id" 1`] = `
<div
  className="row"
>
  <div
    className="col-md-2"
    id="navigation-container"
  >
    <div>
      <div
        id="navigation-logo"
      >
        <h4>
          bui React
        </h4>
      </div>
      <div
        id="navigation-items"
      >
        <ul
          className="navigation"
        >
          <li
            className="bui-react-docs-component-name "
          >
            <a
              onClick={[Function]}
            >
              Getting started
            </a>
          </li>
          <li
            className="bui-react-docs-component-name "
          >
            <a
              onClick={[Function]}
            >
              Theming
            </a>
          </li>
          <li
            className="bui-react-docs-component-name active"
          >
            <a
              onClick={[Function]}
            >
              Button
            </a>
          </li>
          <li
            className="bui-react-docs-component-name "
          >
            <a
              onClick={[Function]}
            >
              Icon
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="col-md-8"
    id="details-container"
  >
    <div
      id="bui-react-docs-component-container"
    >
      <h1
        id="bui-react-docs-component-name"
      >
        Button
      </h1>
      <p
        id="bui-react-docs-component-description"
      >
        Button Component
      </p>
      <br />
      <br />
      <h3
        id="bui-react-docs-component-example-header"
      >
        Example
        s
      </h3>
      <div
        className="bui-react-docs-component-example-container"
      >
        <h4
          className="bui-react-docs-component-example-description"
        >
          Primary/Default Button component
        </h4>
        <button
          data-radium={true}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <span
            data-radium={true}
          >
            Primary Button!
          </span>
        </button>
        <p
          className="bui-react-docs-component-example-code"
        >
          <span
            className="link"
            id="bui-react-docs-component-example-code-toggle-btn"
            onClick={[Function]}
          >
            show
             Code
          </span>
        </p>
        <hr />
      </div>
      <div
        className="bui-react-docs-component-example-container"
      >
        <h4
          className="bui-react-docs-component-example-description"
        >
          Secondary Button component
        </h4>
        <button
          data-radium={true}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <span
            data-radium={true}
          >
            Secondary Button!
          </span>
        </button>
        <p
          className="bui-react-docs-component-example-code"
        >
          <span
            className="link"
            id="bui-react-docs-component-example-code-toggle-btn"
            onClick={[Function]}
          >
            show
             Code
          </span>
        </p>
        <hr />
      </div>
      <div
        className="bui-react-docs-component-example-container"
      >
        <h4
          className="bui-react-docs-component-example-description"
        >
          Button with custom styles
        </h4>
        <button
          data-radium={true}
          style={
            Object {
              "backgroundColor": "#2b95f8",
              "border": "none",
              "color": "#FFFFFF",
              "cursor": "pointer",
              "padding": 20,
              "width": 200,
            }
          }
        >
          <span
            data-radium={true}
          >
            Custom Button!
          </span>
        </button>
        <p
          className="bui-react-docs-component-example-code"
        >
          <span
            className="link"
            id="bui-react-docs-component-example-code-toggle-btn"
            onClick={[Function]}
          >
            show
             Code
          </span>
        </p>
        <hr />
      </div>
      <div
        className="bui-react-docs-component-example-container"
      >
        <h4
          className="bui-react-docs-component-example-description"
        >
          Primary/Default Button component
        </h4>
        <div>
          <button
            data-radium={true}
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
              }
            }
          >
            <span
              data-radium={true}
            >
              Click Me!
            </span>
          </button>
          <br />
          <span>
            Count: 
            <b>
              0
            </b>
          </span>
        </div>
        <p
          className="bui-react-docs-component-example-code"
        >
          <span
            className="link"
            id="bui-react-docs-component-example-code-toggle-btn"
            onClick={[Function]}
          >
            show
             Code
          </span>
        </p>
        <hr />
      </div>
      <div
        className="bui-react-docs-component-example-container"
      >
        <h4
          className="bui-react-docs-component-example-description"
        >
          Link Button component
        </h4>
        <a
          data-radium={true}
          href="#thisisbuttonlink"
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <span
            data-radium={true}
          >
            Link Button!
          </span>
        </a>
        <p
          className="bui-react-docs-component-example-code"
        >
          <span
            className="link"
            id="bui-react-docs-component-example-code-toggle-btn"
            onClick={[Function]}
          >
            show
             Code
          </span>
        </p>
        <hr />
      </div>
      <div
        className="bui-react-docs-component-example-container"
      >
        <h4
          className="bui-react-docs-component-example-description"
        >
          Icon Button component
        </h4>
        <button
          data-radium={true}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <span
            data-radium={true}
          >
            Icon Button
            <span
              data-radium={true}
              style={Object {}}
            >
              <svg
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "  <path d=\\"M4.03 0c-2.53 0-4.03 3-4.03 3s1.5 3 4.03 3c2.47 0 3.97-3 3.97-3s-1.5-3-3.97-3zm-.03 1c1.11 0 2 .9 2 2 0 1.11-.89 2-2 2-1.1 0-2-.89-2-2 0-1.1.9-2 2-2zm0 1c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1c0-.1-.04-.19-.06-.28-.08.16-.24.28-.44.28-.28 0-.5-.22-.5-.5 0-.2.12-.36.28-.44-.09-.03-.18-.06-.28-.06z\\"
                    transform=\\"translate(0 1)\\" />
                  ",
                  }
                }
                data-radium={true}
                height={20}
                style={
                  Object {
                    "fill": "#333",
                  }
                }
                viewBox="0 0 8 8"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              />
            </span>
          </span>
        </button>
        <p
          className="bui-react-docs-component-example-code"
        >
          <span
            className="link"
            id="bui-react-docs-component-example-code-toggle-btn"
            onClick={[Function]}
          >
            show
             Code
          </span>
        </p>
      </div>
      <br />
      <br />
      <h3
        id="bui-react-docs-component-props-header"
      >
        Props
      </h3>
      <div
        id="bui-react-docs-component-props-container"
      >
        <table
          className="props"
        >
          <thead>
            <tr>
              <th>
                Name
              </th>
              <th>
                Description
              </th>
              <th>
                Type
              </th>
              <th>
                Default
              </th>
              <th>
                Required
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td>
                link
              </td>
              <td>
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Specifies if the button can and should act as
                    an anchor. If set to true, then the component
                    will have the same properties of the <i><</i>a<i></i>> tag",
                    }
                  }
                />
              </td>
              <td>
                bool
              </td>
              <td />
              <td>
                Nope
              </td>
            </tr>
            <tr>
              <td>
                icon
              </td>
              <td>
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "",
                    }
                  }
                />
              </td>
              <td>
                string
              </td>
              <td />
              <td>
                Nope
              </td>
            </tr>
            <tr>
              <td>
                text
              </td>
              <td>
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Content to be displayed inside the button.",
                    }
                  }
                />
              </td>
              <td>
                string
              </td>
              <td />
              <td>
                Nope
              </td>
            </tr>
            <tr>
              <td>
                style
              </td>
              <td>
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "This prop can be used to override the default styling of this button component.",
                    }
                  }
                />
              </td>
              <td>
                object
              </td>
              <td>
                {}
              </td>
              <td>
                Nope
              </td>
            </tr>
            <tr>
              <td>
                onClick
              </td>
              <td>
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Callback function fired when the button is clicked.
                    <br/>
                    <pre>
                      <code>
                         function(event: object): void
                      </code>
                      <i>event</i>: Click event targeting the button.
                    </pre>",
                    }
                  }
                />
              </td>
              <td>
                func
              </td>
              <td />
              <td>
                Nope
              </td>
            </tr>
            <tr>
              <td>
                iconStyle
              </td>
              <td>
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Override default styling of the icon.",
                    }
                  }
                />
              </td>
              <td>
                object
              </td>
              <td />
              <td>
                Nope
              </td>
            </tr>
            <tr>
              <td>
                iconWrapperStyle
              </td>
              <td>
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Override the default style of the wrapper around the icon.",
                    }
                  }
                />
              </td>
              <td>
                object
              </td>
              <td />
              <td>
                Nope
              </td>
            </tr>
            <tr>
              <td>
                iconPosition
              </td>
              <td>
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Position of the icon, either <i>\\"right\\"</i> or <i>\\"left\\"</i>",
                    }
                  }
                />
              </td>
              <td>
                enum
              </td>
              <td>
                'right'
              </td>
              <td>
                Nope
              </td>
            </tr>
            <tr>
              <td>
                rounded
              </td>
              <td>
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Can be a \\"number\\" or \\"boolean\\". If boolean and true,
                    the border radius is set to 60.
                    If a number is specified,
                    then the border radius will use that number.",
                    }
                  }
                />
              </td>
              <td>
                union
              </td>
              <td>
                false
              </td>
              <td>
                Nope
              </td>
            </tr>
            <tr>
              <td>
                children
              </td>
              <td>
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Element to be displayed inside the button.
                    If text is defined, then the button will use the value of text
                    in the button over children. But, otherwise, children will be
                    displayed.",
                    }
                  }
                />
              </td>
              <td>
                node
              </td>
              <td />
              <td>
                Nope
              </td>
            </tr>
            <tr>
              <td>
                type
              </td>
              <td>
                <p
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "The type of the button.
                    Can be one of \\"primary\\", \\"secondary\\"",
                    }
                  }
                />
              </td>
              <td>
                enum
              </td>
              <td>
                'primary'
              </td>
              <td>
                Nope
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</div>
`;

exports[`DOCS: App Snapshot Test renders the "Getting Started" page for url "/" 1`] = `
<div
  className="row"
>
  <div
    className="col-md-2"
    id="navigation-container"
  >
    <div>
      <div
        id="navigation-logo"
      >
        <h4>
          bui React
        </h4>
      </div>
      <div
        id="navigation-items"
      >
        <ul
          className="navigation"
        >
          <li
            className="bui-react-docs-component-name active"
          >
            <a
              onClick={[Function]}
            >
              Getting started
            </a>
          </li>
          <li
            className="bui-react-docs-component-name "
          >
            <a
              onClick={[Function]}
            >
              Theming
            </a>
          </li>
          <li
            className="bui-react-docs-component-name "
          >
            <a
              onClick={[Function]}
            >
              Button
            </a>
          </li>
          <li
            className="bui-react-docs-component-name "
          >
            <a
              onClick={[Function]}
            >
              Icon
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="col-md-8"
    id="details-container"
  >
    <div>
      <h2>
        Get Started
      </h2>
    </div>
  </div>
</div>
`;

exports[`DOCS: App Snapshot Test renders the "Theming" page for url "/theming" 1`] = `
<div
  className="row"
>
  <div
    className="col-md-2"
    id="navigation-container"
  >
    <div>
      <div
        id="navigation-logo"
      >
        <h4>
          bui React
        </h4>
      </div>
      <div
        id="navigation-items"
      >
        <ul
          className="navigation"
        >
          <li
            className="bui-react-docs-component-name "
          >
            <a
              onClick={[Function]}
            >
              Getting started
            </a>
          </li>
          <li
            className="bui-react-docs-component-name active"
          >
            <a
              onClick={[Function]}
            >
              Theming
            </a>
          </li>
          <li
            className="bui-react-docs-component-name "
          >
            <a
              onClick={[Function]}
            >
              Button
            </a>
          </li>
          <li
            className="bui-react-docs-component-name "
          >
            <a
              onClick={[Function]}
            >
              Icon
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="col-md-8"
    id="details-container"
  >
    <div>
      <h2>
        Theming
      </h2>
    </div>
  </div>
</div>
`;
